framework/include/utils/Shuffle.h

```diff
@@ -385,6 +419,73 @@ MooseUtils::resample(const std::vector<T> & data,
   return replicate;
 }
 
+template <typename T, typename ActionFunctor>
+void
+MooseUtils::resampleAct(const std::vector<T> & data,
+                        const ActionFunctor & act,
+                        MooseRandom & generator,
+                        const std::size_t seed_index,
+                        const libMesh::Parallel::Communicator * comm_ptr)
```

Comment:
@loganharbour This is the only function that you need to worry about.

Reply:
What a beaut.